sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -833,37 +841,36 @@ private void queueWork(int maximumMessageCount, Duration maxWaitTime,
 
         if (messageSubscriber != null) {
             messageSubscriber.queueWork(work);
-            return;
+            LOGGER.atVerbose().addKeyValue(WORK_ID_KEY, work.getId()).log("Receive request queued up.");
+            return messageSubscriber::processWorkQueue;
         }
 
-        messageSubscriber = synchronousMessageSubscriber.updateAndGet(subscriber -> {
-            // Ensuring we create SynchronousMessageSubscriber only once.
-            if (subscriber == null) {
-                return new SynchronousMessageSubscriber(asyncClient,
+        final boolean isFirstWork;
+        synchronized (createSubscriberLock) {
```

Comment:
The `updateAndGet` is replaced with a `synchronized`. The `updateAndGet` can invoke the compute function more than once under the race.  The `synchronized` adds no overhead here since logic in the block is side-effect free.

Reply:
Note: this is separate from the hang case mentioned in the PR description. But here also we can see another instance of losing work under concurrent pattern.